packages/modules/workflow-engine-redis/src/utils/workflow-orchestrator-storage.ts

```diff
@@ -290,4 +312,42 @@ export class RedisDistributedTransactionStorage extends DistributedTransactionSt
       await job.remove()
     }
   }
+
+  async scheduleJob(
+    jobDefinition: string | { jobId: string },
+    schedulerOptions: SchedulerOptions
+  ): Promise<void> {
+    const jobId =
+      typeof jobDefinition === "string" ? jobDefinition : jobDefinition.jobId
+
+    // In order to ensure that the schedule configuration is always up to date, we first cancel an existing job, if there was one
+    // any only then we add the new one.
+    await this.cancelJob(jobId)
```

Comment:
q: Whats the behavior here when we try to cancel a job thats already running? Does the workflow get gracefully killed or is it a complete rug pull?

Reply:
@riqwan it is not documented, but based on how their repeatable jobs work it should not kill any running job, it will just remove the scheduling. That being said, I guess `cancel` isn't the best verb to use here. I'll need to add a test to be 100% sure though.